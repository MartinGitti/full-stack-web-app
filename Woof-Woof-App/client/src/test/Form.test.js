/*
Please follow these steps in order to run tests properly:

1st: Start backend server: 
In the terminal or command line interface
Type 'nmp start'
Please ensure that you are in the root directory of the project.

2nd: Start Frontend server:
cd to the frontend folder= cd frontend.
Type 'npm start' in frontend directory.

Ensure that both servers are running correctly.

Then cd to test folder in frontend directory.
Type 'npm test' in terminal.
All test files will run and should return passed.
*/

import React from 'react';
import renderer from 'react-test-renderer';

import Form from '../components/Form';

it('renders component correctly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
});
