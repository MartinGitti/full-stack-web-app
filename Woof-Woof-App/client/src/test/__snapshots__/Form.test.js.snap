// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div
  className="form-div"
>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="barking.png"
      />
      <h1>
        Woof Woof Barker 
        <span
          aria-label="dog"
          role="img"
        >
          🐕
        </span>
      </h1>
    </header>
  </div>
  <form
    className="woof-user-form"
  >
    <label>
      Name:
    </label>
    <input
      className="u-full-width"
      name=""
      onChange={[Function]}
      placeholder="Name..."
      type="text"
    />
    <label>
      Woof:
    </label>
    <textarea
      className="u-full-width"
      name=""
      onChange={[Function]}
      placeholder="add your bark here..."
      type="text"
    />
    <button
      className="button"
      onClick={[Function]}
      type="submit"
    >
      Woof Woof Bark 
      <span
        aria-label="dog-face"
        role="img"
      >
        🐶)))
      </span>
    </button>
  </form>
  <div
    className="loading"
  >
    <img
      alt="loading gif"
      className="loading-gif"
      src="loading.gif"
    />
  </div>
</div>
`;
